### PR TITLE
Selected route component

### PR DESCRIPTION
**Pull request by ismailirahim**
Merge

### CODE REVIEW

File: src/client/components/SelectedRouteContainer.js [hunk 1]
```diff
@@ -1,22 +1,87 @@
...
+  componentDidUpdate(_, prevState) {
+    if (prevState.suggested_route_id !== this.state.suggested_route_id) {
+      this.fetchSuggestedRoutes();
+      this.setState({
+        suggested_route_id: null
+      });
+    }
+  }
+  handleSelectRoute = id => {
+    const { vessel_reports_id } = this.props.match.params;
```
Comment (zkwsk) [inline src/client/components/SelectedRouteContainer.js]: Put this in state and read it from there.

File: src/client/components/SelectedRouteContainer.js [hunk 2]
```diff
@@ -1,22 +1,87 @@
...
+  //       error => {
+  //         console.log(error);
+  //       }
+  //     );
+  // };
+  componentDidMount() {
+    this.fetchSuggestedRoutes();
   }
+  fetchSuggestedRoutes = () => {
+    const { voyage_id, vessel_reports_id } = this.props.match.params;
```
Comment (zkwsk) [inline src/client/components/SelectedRouteContainer.js]: Put `voyage_id` in state as well.

File: src/client/components/SuggestedRoutes/SuggestedRouteDetails.js [hunk 3]
```diff
@@ -4,33 +4,33 @@ import Field from "../Field";
...
+        <Field title={"HFO CONSUMPTION"} description={hfo} />
+        <Field title={"MAX WAVE HS"} description={`${maxWaveHs}m`} />
       </div>
       <div>
-        <Field title={"AVG SPEED"} description={avgSpeed} />
-        <Field title={"HIRE COST"} description={hireCost} />
-        <Field title={"CONSUMPTION"} description={consumption} />
+        <Field title={"AVG SPEED"} description={`${avgSpeed}kn`} />
+        <Field title={"HIRE COST"} description={`$${hireCost}`} />
+        <Field title={"LSFO CONSUMPTION"} description={lsfo} />
```
Comment (zkwsk) [inline src/client/components/SuggestedRoutes/SuggestedRouteDetails.js]: Small thing: Please don't uppercase these. It's part of the styling, so just do "Consumption". Exception is Abbreviations, so keep "AVG".

File: src/client/components/SuggestedRoutes/SuggestedRouteSummary.js [hunk 4]
```diff
@@ -6,28 +6,60 @@ export default function SuggestedRouteSummary({
...
+          <Field title="TOTAL COST" description={`$${price}`} />
+          <button onClick={selectRouteBtn}>SELECT ROUTE </button>
+        </>
+      ) : (
+        <>
+          <h2>Selected Route</h2>
+          <Field title="DURATION" description={duration} />
+          <Field title="TOTAL COST" description={`$${price}`} />
+        </>
+      )} */}
```
Comment (zkwsk) [inline src/client/components/SuggestedRoutes/SuggestedRouteSummary.js]: Commented out code

File: src/client/components/SelectedRouteContainer.js [hunk 5]
```diff
@@ -1,22 +1,87 @@
...
+  //     })
+  //     .then(
+  //       response => {
+  //         console.log("this is the response from axios ", response);
+  //       },
+  //       error => {
+  //         console.log(error);
+  //       }
+  //     );
+  // };
```
Comment (zkwsk) [inline src/client/components/SelectedRouteContainer.js]: Commented out code

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js [hunk 6]
```diff
@@ -1,36 +1,39 @@
...
+    const { routeslist, handleSelectRoute } = this.props;
+
     return (
       <div>
-        {this.props.routeslist.map(route => {
+        {routeslist.map(route => {
+          let eta = moment(route.eta);
+          let departureTime = moment(route.departure_time);
+          let duration = moment.duration(eta.diff(departureTime));
+          const d = `${duration.days()} Days ${duration.hours()} Hours`;
```
Comment (zkwsk) [inline src/client/components/SuggestedRoutes/SuggestedRoutesList.js]:
```suggestion
          const durationString = `${duration.days()} Days ${duration.hours()} Hours`;
```

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js [hunk 7]
```diff
@@ -1,36 +1,39 @@
...
-              sailDays={route.sailDays}
-              maxWaveHs={route.maxWaveHs}
-              avgSpeed={route.avgSpeed}
-              hireCost={route.hireCost}
-              consumption={route.consumption}
+              id={route.id}
+              isSelected={route.selected_route_id === route.id}
+              handleClick={handleSelectRoute}
+              key={route.id}
+              duration={d}
```
Comment (zkwsk) [inline src/client/components/SuggestedRoutes/SuggestedRoutesList.js]:
```suggestion
              duration={durationString}
```

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js [hunk 8]
```diff
@@ -43,13 +46,13 @@ SuggestedRoutesList.propTypes = {
   routesList: PropTypes.arrayOf(
     PropTypes.shape({
       id: PropTypes.number.isRequired,
-      isSelected: PropTypes.bool.isRequired,
+      isSelected: PropTypes.bool,
       handleClick: PropTypes.func.isRequired,
       key: PropTypes.number,
```
Comment (zkwsk) [inline src/client/components/SuggestedRoutes/SuggestedRoutesList.js:51]: Remove this

**Review by zkwsk — CHANGES_REQUESTED** [review]

**Review by zkwsk — APPROVED** [review]

**Review by AbodHassam — APPROVED** [review]
Looks good to me

**Pull request merged**